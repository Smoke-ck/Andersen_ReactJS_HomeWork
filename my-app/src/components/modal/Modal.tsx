import {FC} from 'react'
import "./modal.scss"

type IModal ={
  active:boolean
  setActive:any,
  children:any
}
const Modal:FC<IModal>= ({active,setActive,children}) => {
  return(
    <div className={active ? 'modal active' : 'modal'} onClick ={() => {setActive(false)}}>
      <div className="modal__content" onClick={e => e.stopPropagation() }>
        {children}
      </div>
    </div>
  )
}
export default Modal;


