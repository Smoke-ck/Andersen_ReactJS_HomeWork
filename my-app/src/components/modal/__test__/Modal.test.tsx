import Modal, {} from "../Modal"
import { shallow, render, mount } from "enzyme";
import { shallowToJson } from 'enzyme-to-json';

const root = document.createElement('div')
const Child = () => <div>Yolo</div>;

const props = {
  active:false,
  setActive:() => jest.fn(),
  children:root,
  handleToggleModal : () => jest.fn()
}
describe('Modal component', () => {
  const mockfn = jest.fn();
  let wrapper:any;
  beforeEach(() => {
    wrapper = shallow(
      <Modal {...props} handleToggleModal={mockfn} setActive={mockfn}><Child/></Modal>
    );
  })
  it('should render all components ', () => {
    expect(wrapper.find('[className="modal"]').exists()).toBeTruthy();
    expect(wrapper.find('[className="modal__content"]').exists()).toBeTruthy();
    expect(wrapper.find('[className="modal__close"]').exists()).toBeTruthy();
    expect(wrapper.find('[className="deleteMenu__button deleteMenu__button--close"]').exists()).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  });
  it('should toggle when clicked', () => {
    wrapper.find('[className="deleteMenu__button deleteMenu__button--close"]').simulate('click');
    expect(mockfn.mock.calls).toHaveLength(1);
  });
  it('should click close', () => {
    wrapper.find('[className="modal"]').simulate('click');
    expect(mockfn.mock.calls).toHaveLength(1);
  });
});