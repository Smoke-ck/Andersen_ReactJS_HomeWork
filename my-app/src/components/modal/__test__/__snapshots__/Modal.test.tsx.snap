// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render all components  1`] = `
<Portal
  containerInfo={<div />}
>
  <div
    className="modal"
    onClick={[Function]}
  >
    <div
      className="modal__content"
      onClick={[Function]}
    >
      <div
        className="modal__close"
      >
        <button
          className="deleteMenu__button deleteMenu__button--close"
          onClick={[MockFunction]}
        >
          Ð¥
        </button>
      </div>
      <Child />
    </div>
  </div>
</Portal>
`;
